exporter/datadogexporter/metrics/utils.go

```diff
@@ -88,10 +89,27 @@ func addHostname(metrics []datadog.Metric, logger *zap.Logger, cfg *config.Confi
 	}
 }
 
-// addNamespace prepends all metric names with a given namespace
+// shouldPrepend decides if a given metric name should be prepended by `otel.`.
+// By default, this happens for
+// - hostmetrics receiver metrics (since they clash with Datadog Agent system check) and
+// - running metrics
+func shouldPrepend(name string) bool {
+	namespaces := [...]string{"datadog_exporter.", "system.", "process."}
```

Comment:
Nit: I think `datadog_exporter.*` metrics don't ever go through `addNamespace`.

Reply:
They do: they are added in `mapMetrics` and `addNamespace` happens inside `ProcessMetrics`:
 https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d04011af4f5b0d8e12410684629de674ac298d18/exporter/datadogexporter/metrics_exporter.go#L72-L74